tests/parental_controls_gnome_software_oars.pm

```diff
@@ -0,0 +1,112 @@
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+
+    check_desktop_clean();
+
+    # As per parental_controls_setup.pm, we start out with several users and
+    # some parental controls configured. We start out logged in as the
+    # administrator, search for a non-child-friendly game (‘Cow’s Revenge’),
+    # install it and uninstall it. We then switch users to the Shared Account,
+    # which is not an administrator but has no parental OARS restrictions. We
+    # repeat the Cow’s Revenge test, but expecting that installation should
+    # trigger an auth dialogue this time. We then repeat the test with the Tiny
+    # Tim account, who is not an administrator *and* has parental OARS
+    # restrictions. Tiny Tim should not be able to find Cow’s Revenge, or
+    # install it.
+
+    # Run gnome-software.
+    type_very_safely("gnome-software\n");
+    assert_screen('gnome_software_main_screen', 10);
+
+    # Search for Cow’s Revenge. It should appear, since we’re logged in as an
+    # administrator. Cow’s Revenge has some ‘intense’ OARS ratings, but a fairly
+    # small download size, so shouldn’t make the test take too long.
+    type_string('Cow\'s Revenge');
+    assert_and_click('gnome_software_search_cows_revenge', 'left', 10);
+
+    # Install it. We don’t expect an auth dialogue.
+    assert_and_click('gnome_software_cows_revenge_details_uninstalled', 'left', 10);
+    assert_screen('gnome_software_cows_revenge_installing', 10);
+
+    # Wait for installation to succeed.
+    assert_screen('gnome_software_cows_revenge_details_installed', 600);
+    send_key_combo('alt', 'f4');
+
+    # Uninstall it again.
+    $self->uninstall_flatpak_app('com.github.dariasteam.cowsrevenge');
+
+    # Switch user to the Shared Account.
+    $self->switch_user('shared', '');
+
+    # Run gnome-software.
+    type_very_safely("gnome-software\n");
+    assert_screen('gnome_software_main_screen', 10);
+
+    # Search for Cow’s Revenge. It should still appear, since we’re logged in as
+    # the shared user, who has no OARS restrictions set.
+    type_string('Cow\'s Revenge');
+    assert_and_click('gnome_software_search_cows_revenge', 'left', 10);
+
+    # Install it. We expect an auth dialogue this time. Don’t bother checking
+    # for the `gnome_software_cows_revenge_installing` screen, as it often
+    # completes too quickly after the auth dialogue.
+    assert_and_click('gnome_software_cows_revenge_details_uninstalled', 'left', 10);
+    assert_screen('gnome_software_cows_revenge_installing_auth', 10);
+    type_string('123');  # Password
+    send_key('ret');
+
+    # Wait for installation to succeed.
+    assert_screen('gnome_software_cows_revenge_details_installed', 600);
+    send_key_combo('alt', 'f4');
+
+    # Uninstall it again.
+    $self->uninstall_flatpak_app('com.github.dariasteam.cowsrevenge');
+
+    # Switch user to Tiny Tim. Since this is the first time we’ve logged in as
+    # Tiny Tim, his password needs to be set.
```

Comment:
Couldn't you make his account passwordless while creating it to simplify subsequent tests?

Reply:
I could. Partially I did this to incidentally test setting password on first login, and partially to keep all the accounts using the same password (`123`) to keep things as similar as possible in that sense.

Reply:
Fair enough. We might want to add a (separate) test that verifies that passwordless accounts (a) work; (b) do show a Polkit dialog with just an "Authenticate" button (no password field) when you try to do some action that requires authentication. (b) is a downstream behaviour change.

Reply:
Filed https://phabricator.endlessm.com/T25454.